src/components/App.js

```diff
@@ -251,10 +251,13 @@ const App = ({
             )
             const metadata = {
                 [program.id]: program,
-                [programStage.id]: programStage,
                 ...getDynamicTimeDimensionsMetadata(program, programStage),
             }
 
+            if (programStage?.id) {
```

Comment:
if you want to keep it in the initial `metadata` assignment you could do `...(programStage?.id ? { [programStage.id]: programStage } : {})` above. But the separate assignment works too 👍

Reply:
I think it's more clear with a separate assignment.